openedx/core/djangoapps/user_api/views.py

```diff
@@ -345,10 +350,63 @@ def post(self, request):
             }
             return JsonResponse(errors, status=400)
 
+        password = data.get("password")
+        password_copy = data.get("password_copy")
+        if password and password_copy and password != password_copy:
+            errors = {
+                'password_copy': [{"user_message": "Passwords don't match."}]
+            }
+            return JsonResponse(errors, status=400)
+
         response = JsonResponse({"success": True})
         set_logged_in_cookies(request, response, user)
+
+        if getattr(settings, 'REGISTRATION_EMAIL_FULL_VERIFICATION', None) is not None:
+            email = data.get("email")
+            error = self._verify_email_really_exists(email)
+            if error:
+                errors = {
+                    'email': [{"user_message": error}]
+                }
+                return JsonResponse(errors, status=400)
+
         return response
 
+    def _verify_email_really_exists(self, email):
+        """Check if a email really exists,
+        returns None if success, otherwise an error message
+        """
+        mx_record = None
+        try:
+            domain = email.rsplit('@', 1)[-1]
```

Comment:
@schenedx @nedbat Does it make more sense for us to just add another (optionally configurable) email input field to the form, rather than trying to use a validation tool? The root issue is that users are mistyping their emails and having trouble getting registered. We've had a few courses where 3-5% of users are either mistyping their email or password. A double form would not account for users who copy and paste of course, which is how this "email validator" idea came up on our end.